src/state/vaultsLifecycle.ts

```diff
@@ -0,0 +1,429 @@
+      queryKey: ['vaultSlippage', amount, operation],
+      queryFn: async () => {
+        if (operation === 'DEPOSIT' || amount.trim() === '') {
+          return wrapNullable(undefined);
+        }
+        const sharesToWithdraw = MustBigNumber(amount).div(
+          // usdc / share to determine share value
+          MustBigNumber(vaultBalance?.balanceUsdc).div(MustBigNumber(vaultBalance?.balanceShares))
+        );
+        const slippage = await placeholderFetchVaultFormSlippage(
+          sharesToWithdraw.decimalPlaces(0, BigNumber.ROUND_FLOOR).toNumber(),
+          MustBigNumber(amount).times(0.96145).toNumber()
+        );
+
+        const parsedSlippage =
+          VaultDepositWithdrawFormValidator.getVaultDepositWithdrawSlippageResponse(slippage);
+
+        return wrapNullable(parsedSlippage);
+      },
+      ...vaultQueryOptions,
+    });
+    return slippageResp?.data;
+  }
+).dispatchValue((dispatch, value) => {
+  dispatch(setVaultFormSlippageResponse(value ?? undefined));
+});
+
+export const vaultFormValidation = createHookedSelector(
+  [
+    vaultFormAmountDebounced,
+    createAppSelector(
+      [
+        (state) => state.vaults.vaultForm.operation,
+        (state) => state.vaults.vaultForm.slippageAck,
+        (state) => state.vaults.vaultForm.confirmationStep,
+        (state) => state.vaults.vaultForm.slippageResponse,
+      ],
+      (operation, slippageAck, confirmationStep, slippageResponse) => ({
+        operation,
+        slippageAck,
+        confirmationStep,
+        slippageResponse,
+      })
+    ),
+    (state) => state.vaults.vaultAccount,
+    createAppSelector(
+      [
+        (state) => state.account.subaccount?.marginUsage?.current,
+        (state) => state.account.subaccount?.freeCollateral?.current,
+      ],
+      (marginUsage, freeCollateral) => ({
+        marginUsage,
+        freeCollateral,
+      })
+    ),
+  ],
+  (
+    amount,
+    { operation, slippageAck, confirmationStep, slippageResponse },
+    vaultAccount,
+    accountInfo
+  ) => {
+    const { data: validationResponse } = useQueryHf({
+      queryKey: [
+        'vaultFormValidation',
+        amount,
+        operation,
+        slippageAck,
+        confirmationStep,
+        slippageResponse?.toString(),
+        vaultAccount?.toString(),
+        accountInfo,
+      ],
+      queryFn: async () => {
+        const parsedSlippage = VaultDepositWithdrawFormValidator.validateVaultForm(
+          new VaultFormData(
+            operation === 'DEPOSIT'
+              ? VaultFormAction.DEPOSIT
+              : operation === 'WITHDRAW'
+                ? VaultFormAction.WITHDRAW
+                : assertNever(operation),
+            amount != null ? MustBigNumber(amount).toNumber() : undefined,
+            slippageAck,
+            confirmationStep
+          ),
+          new VaultFormAccountData(accountInfo.marginUsage, accountInfo.freeCollateral),
+          vaultAccount,
+          slippageResponse
+        );
```

Comment:
super-nit: also along the lines of readability, params of `vaildateVaultForm` might be easier to understand if it wasn't in-lined.